bu_passport/scripts/eventscraper.py

```diff
@@ -0,0 +1,167 @@
+from datetime import datetime
+
+from selenium import webdriver
+from selenium.webdriver.chrome.service import Service
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.common.exceptions import NoSuchElementException
+import firebase_admin
+from firebase_admin import credentials,firestore
+
+cred = credentials.Certificate("/Users/saisriram/Desktop/BUPassport/serviceAccountKey.json")
+firebase_admin.initialize_app(cred)
+db = firestore.client()
+
+
+import time
+import hashlib
+import random
+
+event_photos = [
+    'image1',
+    'image2',
+    'image3',
+    'image4',
+    'image5',
+    'image6',
+    'image7',
+    'image8',
+    'image9',
+    'image10'
+]
+
+service = Service(executable_path='/Users/saisriram/Desktop/BUPassport/chromedriver')
+driver = webdriver.Chrome(service=service)
+
+driver.get('https://www.bu.edu/arts/calendar/')
+
+WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.CLASS_NAME, 'event-link')))
+
+eventLinks = driver.find_elements(By.CLASS_NAME, 'event-link')
+
+for eventLink in eventLinks:
+    # generate a random number between 0 and 9
+    random_number = random.randint(0, 9)
+
+    random_image = event_photos[random_number]
+
+    eventPhoto = f"assets/images/arts/{random_image}.jpeg"
+
+    event_title_element = eventLink.find_element(By.CSS_SELECTOR, '.event-link a')
+    eventTitle = event_title_element.text.strip()
+    print(eventTitle)
+    event_url = event_title_element.get_attribute('href')
+    print(event_url)
+    driver.get(event_url)    
+
+    WebDriverWait(driver, 5).until(EC.presence_of_element_located((By.CLASS_NAME, 'eventDetail')))
+
+    try:
+        event_detail = driver.find_element(By.CLASS_NAME, 'eventDetail')
+        
+        try:
+            event_tabular = event_detail.find_element(By.CLASS_NAME, 'tabular')
+            try:
+                eventStartTime = event_tabular.find_element(By.XPATH, './dt[text()="Starts:"]/following-sibling::dd').text
+                print("Starts:", eventStartTime)
+            except NoSuchElementException:
+                print("Starts: Not provided")
+
+            try:
+                eventEndTime = event_detail.find_element(By.XPATH, '//dt[text()="Ends:"]/following-sibling::dd').text
+                print("Ends:", eventEndTime)
+            except NoSuchElementException:
+                print("Ends: Not provided")
+            
+            try:
+                eventURL = event_detail.find_element(By.XPATH, '//dt[text()="URL:"]/following-sibling::dd/a').get_attribute('href')
+                print("URL:", eventURL)
+            except NoSuchElementException:
+                print("URL: Not provided")
+            
+            location_room_address = ''
+
+            try:
+                eventRoom = event_detail.find_element(By.XPATH, '//dt[text()="Room:"]/following-sibling::dd').text
```

Comment:
It's fine for now, but since we might consider using the ICS file provided by IS&T, this method might need to be replaced later.